File: src/com/company/Divider.java
package com.company;

public class Divider implements Process {
    public Divider() {
    }

    @Override
    public double process(double a, double b) {
        return a / b;
    }

    @Override
    public int modProcess(int a, int b, int m) {
        if (isPrime(m))
            return (a * inverseElement(b, m) % m);
        else return Integer.MAX_VALUE;
    }

    int binPow(int basa, int p, int m) {
        if (p == 1)
            return basa;
        if (p % 2 == 0) {
            int t = binPow(basa, p / 2, m);
            return (int) (Math.pow(p, 2) % m);
        } else
            return binPow(basa, p - 1, m) * basa % m;
    }

    int inverseElement(int x, int m) {
        return binPow(x, m - 2, m);
    }

    private boolean isPrime(int n) {
        for (int i = 2; i < Math.sqrt(n); i++) {
            if (n % i == 0)
                return false;
        }
        return true;
    }
}

File: ConsoleApp1/Divider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp1
{
    class Divider : IProcess
    {
        public Divider() { }
        public dynamic Process(dynamic A, dynamic B)
        {
            return A / B;
        }
        public int ModProcess(int A, int B, int M)
        {
            if (IsPrime(M))
            {
                return (A * InverseElement(B, M)) % M;
            }
            else return int.MaxValue;
        }
        private int BinPow(int basa, int p, int M)
        {
            if (p == 1)
            {
                return basa;
            }
            if (p % 2 == 0)
            {
                int t = BinPow(basa, p / 2, M);
                return t * t % M;
            }
            else
            {
                return BinPow(basa, p - 1, M) * basa % M;
            }

        }
        private int InverseElement(int x, int M)
        {
            return BinPow(x, M - 2, M);
        }
        private static bool IsPrime(int n)
        {
            for (int i = 2; i <= Math.Sqrt(n); i++)
            {
                if (n % i == 0)
                    return false;
            }
            return true;
        }
    }
}
